### ISSUE TITLE
Widget field depends with type block KO

### ISSUE THREAD

**Issue by Selven99** [magento/magento2#13316]
### Preconditions

1. Magento 2.4-develop
2. PHP 7, MySQL: ANY

### Steps to reproduce

1. Create a custom widget with widget.xml file
2. Create a select option chooser
```xml
<parameter name="test_select" xsi:type="select" visible="true" required="true" sort_order="10">
    <label translate="true">Select</label>
    <options>
        <option name="val_1" value="val_1" selected="true">
            <label translate="true">Value 1</label>
        </option>
        <option name="val_2" value="val_2">
            <label translate="true">Value 2</label>
        </option>
    </options>
</parameter>
```
3. Create a CMS block Chooser that needs the second option selected
```xml
<parameter name="block_2_id" xsi:type="block" visible="true" required="true" sort_order="30">
    <label translate="true">Block 2</label>
    <depends>
        <parameter name="test_select" value="val_2" />
    </depends>
    <block class="Magento\Cms\Block\Adminhtml\Block\Widget\Chooser">
        <data>
            <item name="button" xsi:type="array">
                <item name="open" xsi:type="string" translate="true">Select Block...</item>
            </item>
        </data>
    </block>
</parameter>
```

### Expected result

1. When I select the Val 1, CMS Chooser is not visible
2. When I select the Val 2, CMS Chooser is visible

### Actual result

1. When I select the Val 1, CMS Chooser is visible
2. When I select the Val 2, CMS Chooser is visible

I checked: **/lib/web/mage/adminhtml/form.js**
In trackChange function, I add log after getting target row: `console.log(idTo);` (line 450 ~)

It tries to get item "options_fieldset61f7a8d3475b5def75f144b49f5d69a9_block_2_id" but this selector doesn't exists! If I try $('options_fieldset61f7a8d3475b5def75f144b49f5d69a9_block_2_id') I get NULL.

Generated HTML for CMS chooser is:
```html
<div class="admin__field field field-options_fieldset61f7a8d3475b5def75f144b49f5d69a9_block_2_id  with-addon required _required with-note"
     data-ui-id="widget-instance-edit-tab-properties-fieldset-element-form-field-options-fieldset61f7a8d3475b5def75f144b49f5d69a9-block-2-id">
    <label class="label admin__field-label" for="options_fieldset61f7a8d3475b5def75f144b49f5d69a9_block_2_id"
           data-ui-id="widget-instance-edit-tab-properties-fieldset-element-label-parameters-block-2-id-label"><span>Block 2</span></label>
    <div class="admin__field-control control">
        <div class="admin__field">
            <div class="control-value"></div>
            <label class="widget-option-label"
                   id="options_fieldset61f7a8d3475b5def75f144b49f5d69a9_block_2_ided82f16acfe32d39acd5b9f1d94c5fc2label">Onepage
                Success</label>
            <div id="options_fieldset61f7a8d3475b5def75f144b49f5d69a9_block_2_ided82f16acfe32d39acd5b9f1d94c5fc2advice-container"
                 class="hidden"></div>

        </div>
        <div class="note admin__field-note" id="options_fieldset61f7a8d3475b5def75f144b49f5d69a9_block_2_id-note"></div>
    </div>
</div>
<div class="admin__field field field-chooseroptions_fieldset61f7a8d3475b5def75f144b49f5d69a9_block_2_id  with-addon"
     data-ui-id="widget-instance-edit-tab-properties-fieldset-element-form-field-chooseroptions-fieldset61f7a8d3475b5def75f144b49f5d69a9-block-2-id">
    <label class="label admin__field-label" for="chooseroptions_fieldset61f7a8d3475b5def75f144b49f5d69a9_block_2_id"
           data-ui-id="widget-instance-edit-tab-properties-fieldset-element-note-label"><span></span></label>
    <div class="admin__field-control control">
        <div class="admin__field">
            <div id="chooseroptions_fieldset61f7a8d3475b5def75f144b49f5d69a9_block_2_id"
                 class="control-value admin__field-value"></div>
            <input id="options_fieldset61f7a8d3475b5def75f144b49f5d69a9_block_2_ided82f16acfe32d39acd5b9f1d94c5fc2value"
                   name="parameters[block_2_id]"
                   data-ui-id="widget-instance-edit-tab-properties-element-hidden-parameters-block-2-id" value="2"
                   class="widget-option required-entry" type="hidden">
            <button id="options_fieldset61f7a8d3475b5def75f144b49f5d69a9_block_2_ided82f16acfe32d39acd5b9f1d94c5fc2control"
                    title="Select Block..." type="button" class="action-default scalable btn-chooser"
                    onclick="options_fieldset61f7a8d3475b5def75f144b49f5d69a9_block_2_ided82f16acfe32d39acd5b9f1d94c5fc2.choose()"
                    data-ui-id="widget-button-9">
                <span>Select Block...</span>
            </button>
        </div>
    </div>
</div>
```

Thanks for your help! It seems to be a bug with widget chooser (like CMS block) and depends!

**Comment by Selven99** [comment]
Hello,

Would anyone have information on this problem?

Thanks, :)

**Comment by magento-engcom-team** [comment]
@Selven99, thank you for your report.
We've acknowledged the issue and added to our backlog.

**Comment by jburel-ca** [comment]
Hi,
Do you have any update about this issue?

**Comment by maartenwolfsen** [comment]
Any update on this issue?

**Comment by engcom-Alfa** [comment]
The issue is still reproducible on the fresh 2.4-develop instance

 **Actual Result:**
When I select the Val 1, CMS Chooser is visible
When I select the Val 2, CMS Chooser is visible
![screen](https://user-images.githubusercontent.com/51679138/72155603-1d7fbb80-33bc-11ea-985b-1c349971fcd7.gif)

**Comment by magento-engcom-team** [comment]
@engcom-Alfa Thank you for verifying the issue.

Unfortunately, not enough information was provided to acknowledge ticket. Please consider adding the following:
- [ ] Add `"Component: "` label(s) to this ticket based on verification result. If uncertain, you may follow the best guess

Once all required information is added, please add label `"Issue: Confirmed"` again.
Thanks!

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Alfa 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-30368` were created

**Issue Available**: @engcom-Alfa, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by magento-engcom-team** [comment]
Hi @Selven99. Thank you for your report. 
The issue has been fixed in magento/magento2#30570 by @hostep in 2.4-develop branch
Related commit(s):
 - [e3789f0906f45b565fdb8d13d28c305917cca1bd](https://github.com/magento/magento2/commit/e3789f0906f45b565fdb8d13d28c305917cca1bd)

The fix will be available with the upcoming 2.4.2 release.
